tests/test_operator.py

```diff
@@ -1853,3 +1853,34 @@ def test_issue_1897(self):
             ['t,x0_blk0,y0_blk0,x,y,z', 't,x1_blk0,y1_blk0,x,y,z'],
             't,x0_blk0,y0_blk0,x,y,z,x1_blk0,y1_blk0,x,y,z'
         )
+
+
+class TestInternals(object):
+
+    def test_indirection(self):
+        nt = 10
+        grid = Grid(shape=(4, 4))
+        time = grid.time_dim
+        x, y = grid.dimensions
+
+        f = TimeFunction(name='f', grid=grid, save=nt)
+        g = TimeFunction(name='g', grid=grid)
+
+        idx = time + 1
+        s = Indirection(name='ofs0', mapped=idx)
```

Comment:
not sure I appreciate the need for this indirection class...need to see better

Reply:
unused in OSS, but used in PRO

I however added it here , along with a suitable test, because I foresee its use in OSS as well